src/helper/SM4T_translate.py

```diff
@@ -20,25 +16,59 @@ def check_language_supported_guard(language):
         )
 
 
-def translate_text(text, from_code, to_code):
-    """
-    Translates the given text from the origin language to the target language.
+class Translator:
+    def __init__(self, config: dict):
+        self.device = (
+            f"cuda:{config['device_index']}"
+            if config["translation_device"] == "cuda"
+            else config["translation_device"]
+        )
+        self.tokenizer = SeamlessM4TTokenizer.from_pretrained(
+            config["translation_model"],
+            cache_dir=CONFIG["model_path"],
+        )
+        self.model = SeamlessM4Tv2ForTextToText.from_pretrained(
+            config["translation_model"], cache_dir=CONFIG["model_path"]
+        )
+
+        self.model.config.max_new_tokens = 512
+        self.model.to(self.device)
+
+    def translate_text(self, text, from_code, to_code):
```

Comment:
maybe add type annotations...they are used in most other places within the code base